native/cocos/core/assets/Material.cpp

```diff
@@ -228,6 +228,30 @@ const MaterialPropertyVariant *Material::getProperty(const ccstd::string &name,
     return nullptr;
 }
 
+void Material::prepareInfo(const cc::IMaterialInfo::DefinesType &patch, ccstd::vector<MacroRecord> &curr) {
+    size_t len = _effectAsset != nullptr ? _effectAsset->_techniques[_techIdx].passes.size() : 1;
+    const auto *pOneElement = ccstd::get_if<MacroRecord>(&patch);
```

Comment:
- don't use Hungarian notation
- it is better to use a more meaningful name, i don't know what oneElement mean